engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -241,7 +242,7 @@ public LogChannelInterface getLogChannel() {
     return log;
   }
 
-  public String getXML() {
+  public synchronized String getXML() {
```

Comment:
Once again - this returns "      <slaveserver><cr> " - Nothing at all to synchronize here ... It's a constant.